src/pages/ws/ui/customize-table-modal/columns-list/columns-list.store.ts

```diff
@@ -1,22 +1,17 @@
-import { makeAutoObservable } from 'mobx'
+import { makeAutoObservable, toJS } from 'mobx'
 
 import { IColumns } from '@declarations'
-import columnsStore from '../../../../../store/wsColumns'
+import columnsStore from '@store/wsColumns'
 
 export class ColumnListStore {
-  private columnsToIgnore = new Set(['Gene', 'Variant'])
-
   public get filteredColumns(): IColumns[] {
-    return columnsStore.getExtendedColumns.filter((column: IColumns) =>
-      column.title
-        .toLocaleLowerCase()
-        .includes(columnsStore.searchColumnValue.toLocaleLowerCase()),
-    )
-  }
-
-  public get columnsToDisplay(): IColumns[] {
-    return this.filteredColumns.filter(
-      (column: IColumns) => !this.columnsToIgnore.has(column.title),
+    return columnsStore.getExtendedColumns.filter(
+      (column: IColumns) =>
+        column.title
+          .toLocaleLowerCase()
+          .includes(columnsStore.searchColumnValue.toLocaleLowerCase()) &&
+        column.title !== 'Gene' &&
+        column.title !== 'Variant',
```

Comment:
Probably it would be better to add another prop into `IColumn`, something like `isPermanent: boolean`. But that's a subject for further refactoring.
For now I suggest you to bring back `private columnsToIgnore = ['Gene', 'Variant']` and do `&& !this.columnsToIgnore.includes(column.title)` here.